WalletWasabi/Wallets/Wallet.cs

```diff
@@ -149,31 +149,39 @@ public Node LocalBitcoinCoreNode
 			IFeeProvider feeProvider,
 			CoreNode coreNode = null)
 		{
-			if (State != WalletState.Uninitialized)
+			if (State > WalletState.WaitingForInit)
```

Comment:
Could be better but I noticed a couple of conceptual issues here:

- The Start/Stop lock is managed in WalletManager instead of Wallet.
- Wallet internal state should only be managed inside Wallet and WalletManager should just say Start or Stop.
- Wallet in itself is not thread-safe.
- Can load only one Wallet at a time.

I am planning to sort this out after the release and on that PR this will be addressed too.